### PR TITLE
handle CWL Directory type

### LINKED ISSUE #466: Support CWL Directory type
## Description

Directories could be supported for cases where contents can be retrieved:

- Using S3 buckets, listing all objects under a given directory path.
- Using HTTP, if the server offers some kind of listing.
- Using STAC references (#103)

### PR DESCRIPTION
**Pull request by fmigneault**
## Changes

- Support CWL `Directory` type as `application/directory` Media-Type

## References

- Fixes #466

### CODE REVIEW

**Comment by francisPLT** [comment]
Read through the code as discussed, everything looks good, except that the last commit (1b0e14c) seems to have broken something

I get this error when running tests of trying to launch the application locally.

```
weaver.wps_restapi.colander_extras.SchemaNodeTypeError: Keyword schema 'CWLHintsItem' of type 'oneOf' specification with 'discriminator' attempts to refer to duplicate example values 'InitialWorkDirRequirement' between 'InitialWorkDirRequirementClass' and 'InlineJavascriptRequirementClass'
```

Everything worked perfectly using the previous commit, so as soon as this issue is resolved, this PR is approved.

**Review by francisPLT — APPROVED** [review]
Previously mentioned problem is now fixed.

Feel free to re-request a review after the final failing test is fixed, but I'm approving it for now so you can merge when ready

**Review by francisPLT — APPROVED** [review]

**Comment by fmigneault** [comment]
Failing test in bad4cf9900039f81f2d7b19eec0180dd0cbd03c6 was a one-off HTTP connection error. 
Following commits add more test cases and validations.

**Pull request merged**
